library/src/scala/quoted/runtime/QuoteMatching.scala

```diff
@@ -27,7 +27,7 @@ trait QuoteMatching:
     *  @param pattern `Expr[Any]` containing the pattern tree
     *  @return None if it did not match, `Some(tup)` if it matched where `tup` contains `Expr[Ti]``
     */
-    def unapply[TypeBindings <: Tuple, Tup <: Tuple](scrutinee: Expr[Any])(using pattern: Expr[Any]): Option[Tup]
+    def unapply[TypeBindings, Tup <: Tuple](scrutinee: Expr[Any])(using pattern: Expr[Any]): Option[Tup]
```

Comment:
`TypeBindings` does not appear anywhere in the signature of the method. How is it used?

Reply:
It is used as a placeholder for type-binding definitions. We must create them all in `TypeBindings` to make them available in `Tup`.

Also see https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala#L352-L380
This doc needed an update.

Reply:
What happens if the user manually calls `unapply` and pass a random type to `TypeBindings`?